website/docs/reference/widgets/table/README.md

```diff
@@ -60,183 +114,231 @@ Now, the table widget should be populated with the data coming from the query.
 
-**selectedRow**
+![](</img/pagination_(2)_(2).gif>)
 
-A table you have created might contain a large amount of data. But when you want to get the data of a single row of the table, the selectedRow property comes into play.\
-selectedRow is an internal property that fetches the data of the table row that the user selected. You can bind different widgets to each table row using this property.
+## Server side searching
 
-As you already know by now, you can write JS code anywhere in Appsmith inside \{{\}}., To bind the selected row of the table to another widget, open the property pane of it, and add the code snippet given below:
+Tables come with the capability of server side searching, which is useful for reducing the amount of unnecessary extra results received from queries: rather than requesting a lot of data from the server and then filtering it on the client, this method passes search terms to the server, so that it only needs to fetch data that's relevant. This can significantly improve performance and user experience when working with large data sets.
 
+#### searchText
+
+`searchText` mirrors the text entered in the table's search bar by the user.
+
+To bind a widget using this property, enter the code snippet given below:
+
+```javascript
+{{<table_name>.searchText}}
 ```
-{{<table_name>.selectedRow}}
-```
 
-Where `<table_name>` is the name of your table.
+The following video shows how to bind a text widget to `Table_1` using searchText.
+
+<VideoEmbed host="youtube" videoId="vn6zx7zMeUs" title="searchText" caption="searchText"/>
+
+#### Searching
+
+A search input is available on the table header to filter out records being displayed on the table. You can access the text in the search bar with `Table1.searchText`; anytime that text is changed, the table's `onSearchTextChange` event is triggered. Using the search text and the related event, you can configure your table to query its datasource for the appropriate results:
+
+<VideoEmbed host="youtube" videoId="3ayioaw5uj8" title="How To Setup Server-Side Search For The Table Widget" caption="How To Setup Server-Side Search For The Table Widget"/>
+
+1. Call the API / query with the **onSearchTextChange** event in the table's properties pane.
+2. Pass the value of `Table1.searchText` within the API request / query.
 
-Drag a text widget onto the canvas and bind it to a selected table row of `Table_1`.\
-Open the property pane of the text widget and add the following snippet to its label:
+As a SQL string:
+```sql
+SELECT * FROM users WHERE name LIKE {{"%" + Table1.searchText + "%"}} ORDER BY id LIMIT 10;
+```
 
+As an API request with URL parameters:
 ```
-{{Table_1.selectedRow}}
+https://mock-api.appsmith.com/users?name={{Table1.searchText}}
 ```
 
-<VideoEmbed host="youtube" videoId="0Pl7p1sA1fY" title="selectedRow" caption="selectedRow"/>
+### Server side filtering
```

Comment:
this does not have to be under searching

Reply:
Do you mean it should be at the same heading level as **Server side searching**? I updated them to be equal levels, let me know if you mean something different.